homeassistant/components/automation/numeric_state.py

```diff
@@ -0,0 +1,61 @@
+"""
+homeassistant.components.automation.state
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Offers state listening automation rules.
+"""
+import logging
+
+from homeassistant.helpers.event import track_state_change
+from homeassistant.const import MATCH_ALL
+
+
+CONF_ENTITY_ID = "state_entity_id"
+CONF_BELOW = "state_below"
+CONF_ABOVE = "state_above"
+
+
+def register(hass, config, action):
+    """ Listen for state changes based on `config`. """
+    entity_id = config.get(CONF_ENTITY_ID)
+
+    if entity_id is None:
+        logging.getLogger(__name__).error(
+            "Missing configuration key %s", CONF_ENTITY_ID)
+        return False
+
+    below = config.get(CONF_BELOW)
+    above = config.get(CONF_ABOVE)
+
+    if below is None and above is None:
+        logging.getLogger(__name__).error(
+            "Missing configuration key %s or %s", CONF_BELOW, CONF_ABOVE)
+
+    def numeric_in_range(value, range_start, range_end):
+        """ Checks if value is inside the range
+        :param value:
+        :param range_start:
+        :param range_end:
+        :return:
+        """
+        value = float(value)
```

Comment:
`float` will raise a `ValueError` if it cannot parse the state. Right now the exception will stop further execution of this automation processing this one event. I guess that's fine but it would probably be nicer to print a warning and returning `False` from this method. What do you think?